src/main/java/eu/dissco/backend/service/SpecimenService.java

```diff
@@ -286,23 +304,41 @@ private int getIntParam(String paramName, MultiValueMap<String, String> params,
     }
   }
 
-  private Map<String, List<String>> mapParamsKeyword(MultiValueMap<String, String> params)
+  private <T extends MappingTerm> Map<T, List<String>> mapParamsKeyword(
+      MultiValueMap<String, String> requestParams,
+      Map<String, T> acceptedParams)
       throws UnknownParameterException {
-    var mappedParams = new HashMap<String, List<String>>();
-    for (var entry : params.entrySet()) {
-      var mappedParam = getMappedTerm(entry.getKey());
-      if (mappedParam.isPresent()) {
-        mappedParams.put(mappedParam.get(), entry.getValue());
+    var mappedParams = new HashMap<T, List<String>>();
+    for (var entry : requestParams.entrySet()) {
+      var mappedParam = acceptedParams.get(entry.getKey());
+      if (mappedParam != null) {
+        mappedParams.put(mappedParam, entry.getValue());
       } else {
         throw new UnknownParameterException("Parameter: " + entry.getKey() + " is not recognised");
       }
     }
     return mappedParams;
   }
 
+  public JsonApiWrapper taxonAggregations(MultiValueMap<String, String> params,
```

Comment:
Ok sorry we talked about this last week but I think I didn't explain it well. So all provided info will be used in the query to elastic. Only it will return only the rank below the lowest requested. So if you do this query it will not return anything. If you do `kingdom=plantae&phylum=animalia` it will return all phylum for both kingdoms. I think I would rather keep it this way to keep it generic. It uses all input for filtering, but only returns the rank below the lowest provided.

Reply:
okay if I understand it correctly you have 2 cases: 
1. If 2+ of the same rank, provide aggregation for both ranks. e.g. kingdom=plants and kingdom = animals, provide phyla for both kingdoms
2. if ranks are different, provide aggregation for lowest rank. e.g. kingdom=plants and phyla = worms, provide classes for worms, ignore plants.  

I think we should differentiate between these cases, at least in the logging. if the ranks aren't equal, we can either log a warning ("redundant [higher rank] provided for aggregation") or maybe throw an exception.

Reply:
Maybe that can be a new PR though.

Reply:
Discussed